libsolidity/analysis/TypeChecker.cpp

```diff
@@ -356,7 +356,15 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 			if (!type(var)->canLiveOutsideStorage() && _function.isPublic())
 				m_errorReporter.typeError(var.location(), "Type is required to live outside storage.");
 			if (_function.isPublic() && !(type(var)->interfaceType(isLibraryFunction)))
-				m_errorReporter.fatalTypeError(var.location(), "Internal or recursive type is not allowed for public or external functions.");
+			{
+				std::string msg;
+				if (isLibraryFunction)
+					msg = "Internal type is not allowed for public or external library functions.";
```

Comment:
If I want to pass a recursive struct in memory to a library, it should not be recursive either - isn't it?